tests/unittests/graphOptzTest.cpp

```diff
@@ -1169,3 +1169,42 @@ TEST_F(GraphOptz, VarsCSE) {
   ASSERT_TRUE(input3->getUsers().size() == 1);
   EXPECT_TRUE(input3->getUsers().begin()->getUser() == RN);
 }
+
+// Verify that constant input canonicalization works correctly when the
+// arithmetic nodes have multiple users.
+TEST_F(GraphOptz, simplifyArithmetic) {
+  Node *I1 = mod_.createVariable(ElemKind::FloatTy, {10, 10, 10}, "input1",
+                                 VisibilityKind::Public, false);
+
+  Type t(ElemKind::FloatTy, {10, 10, 10});
+  Node *SN = F_->createSplat("one", &t, 1.0);
+
+  // The splat is a constant input to add1 and add2, and is their LHS input. We
+  // expect canonicalization to occur during optimization, moving the splat
+  // to the RHS for both. Note that add1 has multiple users: add2 and save1.
+  Node *AN1 = F_->createAdd("add1", SN, I1);
+  Node *AN2 = F_->createAdd("add2", SN, AN1);
+  F_->createSave("save1", AN1);
+  F_->createSave("save2", AN2);
+
+  // Five nodes in total: one splat, two adds, and two saves.
+  EXPECT_EQ(F_->getNodes().size(), 5);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::SplatNodeKind), 1);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::AddNodeKind), 2);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::SaveNodeKind), 2);
+
+  // input1 should have a single user before and after optimization.
+  EXPECT_EQ(I1->getUsers().size(), 1);
+
+  // Simplify nodes will canonicalize add1, and should replace all its users.
+  ::glow::optimize(F_, CompilationMode::Infer);
+
+  // We should have the same five nodes: one splat, two adds, and two saves.
+  EXPECT_EQ(F_->getNodes().size(), 5);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::SplatNodeKind), 1);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::AddNodeKind), 2);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::SaveNodeKind), 2);
```

Comment:
This doesn't check that the expected canonicalization happened.
Could you check that the splat is indeed on the left hand side?
We should also check that the shape of the graph is still what we want (i.e., save1 takes add1, etc.)

Reply:
This test isn't actually meant to verify canonicalization works -- it's meant to make sure that canonicalization does not accidentally duplicate a node for each user.  I will still check to make sure everything else is expected.